### ISSUE TITLE
Move `shared` tests to separate files

### ISSUE THREAD

**Issue by sunfishcode** [WebAssembly/threads#187]
Currently, this threads proposal repo adds `shared` memory tests to existing test files in test/core: memory.wast, imports.wast, and exports.wast. For example:

https://github.com/WebAssembly/threads/blob/main/test/core/memory.wast#L9

However, not all wasm engines are expected to implement this threads proposal in its current form, even once it's standardized, some because of the nondeterminism, and some because of the effective dependency on Web Worker-style threads.

Would it make sense to move the tests for `shared` memory into separate files, so that engines which don't implement `shared` memories can easily filter them? This would be similar to the SIMD tests, which are all named `simd_*.wast`.

**Comment by rossberg** [comment]
I agree, and would go even further. As with the simd tests, we should have a subdirectory `threads`, and all related tests should go there. Similarly for other non-trivial and possibly profile-dependent feature proposals, like exceptions, gc, continuations.
